### PR TITLE
Upgrade operator-sdk to 1.28.0

### PR DESCRIPTION
**Pull request by roypaulin**
Changes reflect migration steps as outlined here:
https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.28.0/

prior version was v1.25.2

Go version moved to 1.19.6
golanci-lint version moved to 1.51.1

### CODE REVIEW

File: helm-charts/verticadb-operator/values.yaml [hunk 1]
```diff
@@ -31,7 +31,7 @@ image:
 
 rbac_proxy_image:
   repo: gcr.io
-  name: kubebuilder/kube-rbac-proxy:v0.13.0
+  name: kubebuilder/kube-rbac-proxy:v0.13.1
```
Comment (spilchen) [inline helm-charts/verticadb-operator/values.yaml:34]: There is an update missing in `tests/external-images-common-ci.txt`

**Review by spilchen — COMMENTED** [review]
Just a few comments...

**Comment by spilchen** [comment]
Did we need to make this change?

```

    [gov4alpha] For scaffolds, replace scaffoldings of v1beta1 by using k8s.io/api/admission/v1 instead of k8s.io/api/admission/v1beta1 as v1beta1 is removed in latest k8s.

  - admissionv1beta1 "k8s.io/api/admission/v1beta1"
  + admissionv1 "k8s.io/api/admission/v1"
```
https://v1-28-x.sdk.operatorframework.io/docs/upgrading-sdk-version/v1.28.0/

We don't use the gov4 plugin. But is this needed to run the operator on k8s 1.26?

**Comment by roypaulin** [comment]
> Did we need to make this change?
> 
> ```
> 
>     [gov4alpha] For scaffolds, replace scaffoldings of v1beta1 by using k8s.io/api/admission/v1 instead of k8s.io/api/admission/v1beta1 as v1beta1 is removed in latest k8s.
> 
>   - admissionv1beta1 "k8s.io/api/admission/v1beta1"
>   + admissionv1 "k8s.io/api/admission/v1"
> ```
> 
> https://v1-28-x.sdk.operatorframework.io/docs/upgrading-sdk-version/v1.28.0/
> 
> We don't use the gov4 plugin. But is this needed to run the operator on k8s 1.26?

Yes I read some doc on this and `k8s.io/api/admission/v1beta1` has been deprecated for a while (since v1.19). Maybe we should just upgrade to v1? Any reason to stick to v1beta1?

**Comment by spilchen** [comment]
No reason to stick with v1beta1. The minimum k8s version we officially support is 1.21.1+. So, lets change it.

**Review by spilchen — APPROVED** [review]

**Pull request merged**